tests/integration.rs

```diff
@@ -0,0 +1,433 @@
+    let client = Box::new(pebble.http_client());
+    let (account, _) = Account::create_with_http(
+        &NewAccount {
+            contact: &[],
+            terms_of_service_agreed: true,
+            only_return_existing: false,
+        },
+        &pebble.directory_url(),
+        None,
+        client,
+    )
+    .await?;
+
+    // Create an order.
+    let identifier = Identifier::Dns("example.com".to_string());
+    let order = account
+        .new_order(&NewOrder {
+            identifiers: &[identifier],
+        })
+        .await?;
+
+    // Issue a certificate w/ HTTP-01 challenge and verify it.
+    test_http1(&pebble, order).await?;
+
+    Ok(())
+}
+
+async fn test_http1(env: &PebbleEnvironment, mut order: Order) -> Result<(), Box<dyn StdError>> {
+    let authorizations = order.authorizations().await?;
+    let mut challenges = Vec::with_capacity(authorizations.len());
+    let mut names = Vec::with_capacity(authorizations.len());
+
+    // Collect up the relevant challenges, provisioning the expected responses as we go.
+    for authz in &authorizations {
+        match authz.status {
+            AuthorizationStatus::Pending => {}
+            AuthorizationStatus::Valid => continue,
+            _ => unreachable!("unexpected authz state: {:?}", authz.status),
+        }
+
+        let challenge = authz
+            .challenges
+            .iter()
+            .find(|c| c.r#type == ChallengeType::Http01)
+            .ok_or_else(|| "no http01 challenge found")?;
+
+        let Identifier::Dns(identifier) = &authz.identifier;
+
+        // Provision the HTTP-01 challenge response with the Pebble challenge test server.
+        let key_auth = order.key_authorization(challenge);
+        env.add_http01_response(&challenge.token, key_auth.as_str())
+            .await?;
+
+        challenges.push((identifier, &challenge.url));
+        names.push(identifier.clone());
+    }
+
+    // Tell the CA we have provisioned the response for each challenge.
+    for (_, url) in &challenges {
+        order.set_challenge_ready(url).await?;
+    }
+
+    // Poll until the order is ready.
+    poll_until_ready(&mut order).await?;
+
+    // Issue a certificate for the names and test the chain validates to the issuer root.
+    issue_certificate(&mut order, names, env.issuer_roots().await?).await
+}
+
+// Poll the given order until it is ready, waiting longer between each attempt up to
+// a maximum.
+//
+// Returns an error when the maximum number of tries has been reached.
+async fn poll_until_ready(order: &mut Order) -> Result<(), Box<dyn StdError>> {
+    let mut tries = 1u8;
+    let mut delay = Duration::from_millis(250);
+    loop {
+        sleep(delay).await;
+        let state = order.refresh().await.unwrap();
+        if let OrderStatus::Ready | OrderStatus::Invalid = state.status {
+            break;
+        }
+
+        delay *= 2;
+        tries += 1;
+        match tries < 10 {
+            true => eprintln!(
```

Comment:
Maybe good to set this up with a simple `tracing_subscriber` like

```rust
    tracing_subscriber::fmt()
        .with_env_filter(tracing_subscriber::EnvFilter::from_default_env())
        .init();
```

and use `tracing::info!()` calls instead of `eprintln!()` here?